UnitTests/Tests/Controllers/TRApplicationControllerSpec.m

```diff
@@ -61,6 +62,44 @@ @interface TRWeatherController (Tests)
             OCMVerify([updateWeatherCommand execute:applicationController]);
         });
     });
+
+    describe(@"subscribeForNotificationsWithDeviceToken:", ^{
+        it(@"subscribes to notifications on the channel corresponding to the device's time zone", ^{
+            TRCourierClient *courier = OCMClassMock([TRCourierClient class]);
+            TRApplicationController *applicationController = [TRApplicationController new];
+            applicationController.courier = courier;
+
+            NSData *deviceToken = [NSData new];
+            [applicationController subscribeForNotificationsWithDeviceToken:deviceToken];
+
+            NSString *channel = [[NSTimeZone localTimeZone] name];
+            OCMVerify([courier subscribeToChannel:channel withToken:deviceToken]);
+        });
+
+        it(@"unsubscribes from notifications if the device's time zone has changed", ^{
+            TRCourierClient *courier = OCMClassMock([TRCourierClient class]);
+            TRApplicationController *applicationController = [TRApplicationController new];
+            applicationController.courier = courier;
+
+            NSData *deviceToken = [NSData new];
+
+            NSTimeZone *initialTimeZone = [NSTimeZone timeZoneWithAbbreviation:@"CEST"];
+            [NSTimeZone setDefaultTimeZone:initialTimeZone];
+            [applicationController subscribeForNotificationsWithDeviceToken:deviceToken];
+
+            [NSTimeZone setDefaultTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"HKT"]];
+            [applicationController subscribeForNotificationsWithDeviceToken:deviceToken];
+
+            OCMVerify([courier unsubscribeFromChannel:initialTimeZone.name]);
+        });
+    });
 });
 
 QuickSpecEnd
+
+void inTimeZone(NSTimeZone *timeZone, void (^callback)(void)) {
```

Comment:
It is now. I wanted to make sure any time zone changes didn't impact other tests. I'm not sure if it's actually a problem, but I hope this function prevents it either way. Plus it reads kind of nice.